x/evm/module.go

```diff
@@ -390,6 +390,31 @@ func (am AppModule) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) []abci.V
 			}
 		}
 	}
-	am.keeper.SetBlockBloom(ctx, utils.Map(evmTxDeferredInfoList, func(i *types.DeferredInfo) ethtypes.Bloom { return ethtypes.BytesToBloom(i.TxBloom) }))
+	allBlooms := utils.Map(evmTxDeferredInfoList, func(i *types.DeferredInfo) ethtypes.Bloom { return ethtypes.BytesToBloom(i.TxBloom) })
+	evmOnlyBlooms := make([]ethtypes.Bloom, 0, len(evmTxDeferredInfoList))
+	for _, di := range evmTxDeferredInfoList {
+		if len(di.TxHash) == 0 {
+			continue
+		}
+		r, err := am.keeper.GetReceipt(ctx, common.BytesToHash(di.TxHash))
```

Comment:
i think we need to use GetTransientReceipt here since receipts haven't been flushed yet